### PR TITLE
[ARCTIC-907][SPARK]Duplicate keys verification for the source table should add a property to control in Spark

### LINKED ISSUE #907: [Improvement]: Duplicate keys verification for the source table should add a property to control in Spark
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

In Spark SQL, e.g. insert into, insert overwrite, CTAS. We will check whether the primary key is not duplicated in the source table.
I think duplicate keys verification for the source table should add a property to control

### How should we improve?

Add a spark property, e.g. spark.sql.check-data-duplicates.enabled 
the default value is true

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

resolve #907 

## Brief change log

Add a spark property: spark.sql.check-data-duplicates.enabled, the default value is true 

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java [hunk 1]
```diff
@@ -31,4 +31,8 @@ public class SparkSQLProperties {
   public static final String REFRESH_CATALOG_BEFORE_USAGE = "spark.sql.arctic.refresh-catalog-before-usage";
 
   public static final String REFRESH_CATALOG_BEFORE_USAGE_DEFAULT = "false";
+
+  public static final String CHECK_DATA_DUPLICATES_ENABLE = "spark.sql.check-data-duplicates.enabled";
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java:35]: Does `spark.sql.arctic.check-source-data-uniqueness.enabled` property name is more appropriate？

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java [hunk 2]
```diff
@@ -31,4 +31,8 @@ public class SparkSQLProperties {
...
+
+  public static final String CHECK_DATA_DUPLICATES_ENABLE_DEFAULT = "true";
```
Comment (zhoujinsong) [inline spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java:37]: Should we change default value to `false`,  considering the big cost of this operation?

**Review by zhoujinsong — COMMENTED** [review]
@hellojinsilei  I left some comments.

File: spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java (hunk 2 above)
Comment (hellojinsilei) [reply spark/v3.1/spark/src/main/java/com/netease/arctic/spark/SparkSQLProperties.java:37]: If we want to make the primary key unique by default, does it means this parameter should also be enabled by default?

**Comment by zhoujinsong** [comment]
You may need to add this new configuration to the user documentation. @hellojinsilei

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**